spec/spec_helper.rb

```diff
@@ -48,4 +52,45 @@ def with_memory_logger level = nil
       Asciidoctor::LoggerManager.logger = old_logger
     end
   end
+
+  def compare_images reference, actual, difference = nil
+    diff = []
+    images = [ChunkyPNG::Image.from_file(reference), ChunkyPNG::Image.from_file(actual)]
+
+    images.first.height.times do |y|
+      images.first.row(y).each_with_index do |pixel, x|
+        diff << [x,y] unless pixel == images.last[x,y]
+      end
+    end
+
+    if diff.length > 0 and difference
+      x, y = diff.map{ |xy| xy[0] }, diff.map{ |xy| xy[1] }
+      images.last.rect(x.min, y.min, x.max, y.max, ChunkyPNG::Color.rgb(0,255,0))
+      images.last.save(difference)
+    end
+
+    diff.length
+  end
+
+end
+
+RSpec::Matchers.define(:match_fixture) do |reference_filename|
```

Comment:
match_fixture seems too generic of a name. It should be something like "has_visual_match" or "visually_matches".